Feature/ab/addbook

AnastasiaBologan opened the pull request.

Zerdados commented on src/components/Overview.tsx:
```diff
@@ -27,30 +24,32 @@ export default function Overview() {
 
         fetchData();
-    }, [location]);
+    }, [setBooks]);  // Abhängigkeitsarray korrigiert
```
We should remove this comment. Seems to only explain why something was changed, which is usefull now, but will be confusing later.

Zerdados commented on src/components/Overview.tsx:
```diff
@@ -27,30 +24,32 @@ export default function Overview() {
                         className="flex-item"
-                        book={book}
+                        onClick={(event: React.MouseEvent<HTMLDivElement, MouseEvent>) => {
+                            event.preventDefault();  // Optional, verhindert das Standardverhalten
```
Why do we need to prevent the default behaviour here?

Zerdados reviewed (changes requested):
I like the new book creation, it looks nice and works nice.

However, I have a suggestion regarding the implementation: Currently, when a book has been successfully created in the backend, we modify the state in the frontend with the data we have sent to the backend. 
You have achieved this by moving the books state from the Overview component to the App component, and adding an addBook function to add new books to that state.

This works fine for our example, but it seems overly complicated and might lead to issues later (e.g. what if the backend were to modify the values we sent it somehow? Now the book that was created in the backend and the book we added to our frontend state are not the same :c ) 

My suggestion would be that you revert the changes you have made to the frontend state (e.g. Everything in `App.tsx` and some of the stuff in `Overview.tsx` and `AddBook.tsx`).
In order to ensure that our newly created books are still displayed when we return to the overview page after the creation, add `{ state: { refresh: true } }` to the `navigate()` command in `AddBook.tsx`.

AnastasiaBologan commented:
Zusammenfassung von Änderungen, danke fürs Feedback, ich hoffe, ich habe es richtig verstanden: Auf Englisch extra für dich

Simplified State Management: Removed the books state from App.tsx, relying on the backend for book data.

Improved Navigation: Added { state: { refresh: true } } to navigation after creating a book to ensure updated data in the overview.

Code Enhancements: Fixed label-input associations and removed unnecessary complexity in AddBook and Overview.

Error Handling: Improved API error handling in the AddBook form.

Zerdados reviewed (approved):
Looks good 🎉
